src/hb-ot-layout.hh

```diff
@@ -314,13 +314,17 @@ _hb_glyph_info_get_unicode_space_fallback_type (const hb_glyph_info_t *info)
 	 hb_unicode_funcs_t::NOT_SPACE;
 }
 
+static inline bool _hb_glyph_info_substituted (const hb_glyph_info_t *info);
 static inline bool _hb_glyph_info_ligated (const hb_glyph_info_t *info);
+static inline bool _hb_glyph_info_multiplied (const hb_glyph_info_t *info);
 
 static inline bool
 _hb_glyph_info_is_default_ignorable (const hb_glyph_info_t *info)
 {
   return (info->unicode_props() & UPROPS_MASK_IGNORABLE) &&
-	 !_hb_glyph_info_ligated (info);
+	 !_hb_glyph_info_substituted (info) &&
+	 !_hb_glyph_info_ligated (info) &&
+	 !_hb_glyph_info_multiplied (info);
```

Comment:
I’m not entirely sure the last two checks are actually needed, in my limited tests whenever `_hb_glyph_info_ligated()` or `_hb_glyph_info_multiplied()` return `true`, `_hb_glyph_info_substituted()` return `true` as well (but the reverse is not true).

Reply:
Correct. Let's do that.

Reply:
LGTM otherwise.